problems/39.Design HashMap.gs

```diff
@@ -0,0 +1,128 @@
+class MyNode {
+    key: number
+    val: number
+    next: MyNode | null
+
+    constructor(key: number, value: number, nextElem: MyNode | null) {
+        this.val = value
+        this.next = nextElem
+        this.key = key
+    }
+}
+
+function add(head: MyNode, key: number, val: number) {
+    let cur = head
+
+    while (cur.next != null) {
+        cur = cur.next
+    }
+    cur.next = new MyNode(key, val, null)
+}
+
+function print(head: MyNode | null) {
+    let cur: MyNode | null = head
+    while (cur != null) {
+        console.log(cur.key, cur.val)
+        cur = cur.next
+    }
+}
+
+function find(head: MyNode | null, key: number): number {
+    let cur: MyNode | null = head
+    while (cur != null) {
+        if (cur.key == key) {
+            return cur.val
+        }
+        cur = cur.next
+    }
+    return -1
+}
+function remove(head: MyNode, key: number): MyNode | null {
+    if (head.key == key) {
+        let newHead = head.next
+        head.next = null //good practise so memory can be free
+        return newHead
+    }
+
+    let cur: MyNode | null = head
+    while (cur.next != null) {
+        if (cur.next.key == key) {
+            break
+        }
+        cur = cur.next
+    }
+    if (cur.next != null) {
+        let removed = cur.next
+        cur.next = cur.next.next
+        removed.next = null //for memory good practise to do, not related to logic
+    }
+    return head
+}
+function findNode(head: MyNode | null, key: number): MyNode | null {
+    let cur: MyNode | null = head
+    while (cur != null) {
+        if (cur.key == key) {
+            return cur
+        }
+        cur = cur.next
+    }
+    return null
+}
+
+
+class MyHashMap {
+    head: MyNode | null = null
+
+    constructor() {
+        this.head = null
+    }
+
+    put(key: number, value: number): void {
+        if (this.head != null) {
+            if (this.get(key) == -1) {
```

Comment:
It;s wasteful to use this.get just to check and then latter find. Doing same work 2x. 

Instead of this.get before if use let node = findNode(... and then check to see if you have null then add if not do update...